From 367c3be2375f7ace18beea4c6616b5f53efa5f29 Mon Sep 17 00:00:00 2001
From: Tang Yuantian <Yuantian.Tang@nxp.com>
Date: Thu, 12 Jan 2017 10:16:52 +0800
Subject: [PATCH 245/388] clk: qoriq: add pll clock to clock lookup table

Registered each PLL clock and its division clock to clock
lookup table to facilitate the clock look up.

Signed-off-by: Tang Yuantian <yuantian.tang@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/clk/clk-qoriq.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/clk/clk-qoriq.c b/drivers/clk/clk-qoriq.c
index 3f1df32..74db468 100644
--- a/drivers/clk/clk-qoriq.c
+++ b/drivers/clk/clk-qoriq.c
@@ -12,6 +12,7 @@
 
 #include <linux/clk.h>
 #include <linux/clk-provider.h>
+#include <linux/clkdev.h>
 #include <linux/fsl/guts.h>
 #include <linux/fsl/svr.h>
 #include <linux/io.h>
@@ -1059,6 +1060,7 @@ static void __init create_one_pll(struct clockgen *cg, int idx)
 
 	for (i = 0; i < ARRAY_SIZE(pll->div); i++) {
 		struct clk *clk;
+		int ret;
 
 		snprintf(pll->div[i].name, sizeof(pll->div[i].name),
 			 "cg-pll%d-div%d", idx, i + 1);
@@ -1072,6 +1074,11 @@ static void __init create_one_pll(struct clockgen *cg, int idx)
 		}
 
 		pll->div[i].clk = clk;
+		ret = clk_register_clkdev(clk, pll->div[i].name, NULL);
+		if (ret != 0)
+			pr_err("%s: %s: register to lookup table failed %ld\n",
+				__func__, pll->div[i].name, PTR_ERR(clk));
+
 	}
 }
 
-- 
2.9.3

